package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +582,46 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {attr: i for i, attr in enumerate(attrs)}  # column numbers
+        keys = col_ids.keys()
+        # check for ids and what type of coordinate convention
+        ids = "id" in keys
+        # coord_dict keys = convention keys = column data or False
+        coord_dict = {conv: False for conv in self._conventions[1:]}
+        for coordtype, col_names in self._coordtype_column_names.items():
+            if all(axis in col_ids for axis in col_names):
+                coord_dict[coordtype] = [col_ids[axis] for axis in col_names]
```

Comment:
This seems quite convoluted to me, and I suspect @IAlibay, given his question. Perhaps you could simplify it and add more explicit variable names, and define the variables where they are locally used? You could only populate the dictionary for column indices if they are found, instead of setting them to False to start with. e.g. (just a suggestion):

```python
attr_to_col_ix = {x: i for i, x in enumerate(attrs)}
convention_to_col_ix = {}
for cv_name, cv_col_names in self._coordtype_column_names.items():
    try:
        convention_to_col_ix[cv_name] = [attr_to_col_ix[x] for x in cv_col_names]
    except KeyError:
        pass
if self.lammps_coordinate_convention == "auto":
    try:
        # this will automatically select in order of priority
        # as defined in the original coordtype_column_names dict
        self.lammps_coordinate_convention = list(convention_to_col_ix)[0]
    except IndexError:
        raise ValueError("No coordinate information detected")
elif not self.lammps_coordinate_convention in convention_to_col_ix:
    raise ValueError(f"No coordinates following {self.lammps_coordinate_convention} found in timestep")

coord_cols = convention_to_col_ix[self.lammps_coordinate_convention]
```